app/models/intervention.rb

```diff
@@ -510,6 +511,13 @@ def update_temporality
     end
   end
 
+  # Set product name if its blank
+  def set_product_name(product,index)
+    land_parcel_name = self.product_parameters.where(type: "InterventionTarget", reference_name: "land_parcel")[index].product.name rescue ""
+    culture_name = self.product_parameters.where(:type=> "InterventionOutput",reference_name: "plant")[index].variant.name rescue ""
```

Comment:
In such a case pass in the output as a parameter instead of relying on indexes.

Reply:
@Aquaj ,

Here Multiple land parcel and culture present, Whose new name field are blanked then we have to update the name based on land parcel and culture name.

Reply:
See my answer to the previous comment. I detail the structure and how you can avoid to have to do so.